### DISCUSSION TITLE
HPC Community Tutorials

### DISCUSSION THREAD

**Discussion by alansill** [hpc-social/hpc-social.github.io#48] [Ideas]
Food for thought: https://www.youtube.com/@hpc.social

We could host tutorials produced by the community for various HPC software, hardware, or features. This channel could be their home. Anything people would be willing to provide to us with a suitable license could be included.

**Comment by vsoch** [comment]
Love that idea! I've definitely had videos that I've shared that I just throw on my personal account (talks, or my SC dance) and it would be fun to have a community place to put it.

**Comment by glennklockwood** [comment]
A lot of tutorials/talks I've given that were recorded belong to the conference or org that hosted me. Creating collections (playlists?) that aggregate tutorials from across the various already-existing online resources might be a lower barrier, and it avoids having to deal with copyright/redistribution rights/etc.

Relatedly, when I saw this subject, I was thinking it was more about written tutorials.  I've made lots of those over the years, but maybe blog syndication already captures some of that.

**Reply by vsoch** [reply]
If needed, I do know a way to render a preview for links, e.g., see "Bash Tutorials" in this page here: https://rse-ops.github.io/knowledge/docs/programming-languages/bash#bash-tutorials. I could imagine a simple page that is entirely made up of those cards, organized by category or similar, and rending from a yaml list in the repo.

**Reply by hearnsj** [reply]
As we know there is a skills gap in HPC. I work in professional services, configuring and deploying HPC systems.
From my point of view there are lots of tutorial resources for programming. I am a member of the Julia community and you can find high quality tutorials from beginner to expert on Youtube.
However for system administration and system installation there are few resources. We are still in the mode of seeing postdoc scientists co-opted into being systems admins for a project and growing into the career from there.
I think there is definite scope for tutorials on:

* Introductory Linux sysadmin
* Cluster management frameworks
* Security hardening

The last topic is going to see much more activity IMHO

**Reply by hearnsj** [reply]
I would also suggest hooking up with the Dell HPC Community run by Jay Boisseau and Janet Morss.
The weekly seminar series are not Dell specific in many instances - for instance the recent excellent Spack tutorial.

**Reply by alansill** [reply]
Good points. We could do both: link to existing tutorials that are accessible on various topics, and provide links to ones that we host for those that do not yet have a home!

**Reply by vsoch** [reply]
I think you're spot on - I think the best advice for those topics I've seen is on the HPC reddit! https://www.reddit.com/r/HPC/

I always say you can measure the life maturity of someone by whether their browse autocompletes to "reddit" or "redfin." I'm in the first bucket. :laughing:

**Reply by vsoch** [reply]
We have to be careful here linking to vendor resources that we aren't presenting biased material.

**Reply by vsoch** [reply]
@alansill that reminds me - you should ping Julie Ma at Ask.CI https://ask.cyberinfrastructure.org/ to see if there are any collaboration ideas. I tried to get US-RSE interested in the various working groups but nobody really was. For some background, Ask.CI is a question/answer sort of deal for our cyberinfrastructure communities (including HPC). I don't have strong ideas at the moment, but wanted to throw it out there!

**Reply by alansill** [reply]
Sure. I'm a bit saturated right now in terms of time for communications so please feel free to reach out on your own.

**Reply by hearnsj** [reply]
I must say I am not a Reddit person. 
I agree that we should not link to vendor resources that are pushing products. However let's say that Bright have introductory courses on the Nvidia Partner training portal. I think that is fair game.

**Reply by vsoch** [reply]
I think reddit is my second favorite social network... after GitHub! I like the content and variety between https://www.reddit.com/r/LongBoi/ and https://www.reddit.com/r/HPC and I think I look over just about everything on https://www.reddit.com/r/Programming/. The only drawback to reddit is that people can be a bit mean at times, but not bad if you expect it.

But now I'm curious! When you type "red" into your browser, what does it autocomplete to?

**Reply by sramkrishna** [reply]
Crotchety people for sure. I mod a 75k community on reddit and it's mostly cool. The /r/linux can get very ugly. I'm somewhat appalled at times though.

What about stackoverflow?

**Reply by vsoch** [reply]
Stackoverlfow is really strict about posts. If you want to start your own community, it's a really long (hard) process too - ask.ci tried for the cyberinfrastructure community (and failed).

**Reply by mtwest2718** [reply]
> @alansill that reminds me - you should ping Julie Ma at Ask.CI https://ask.cyberinfrastructure.org/ to see if there are any collaboration ideas. I tried to get US-RSE interested in the various working groups but nobody really was. For some background, Ask.CI is a question/answer sort of deal for our cyberinfrastructure communities (including HPC). I don't have strong ideas at the moment, but wanted to throw it out there!

Hi @vsoch. Is this still a live project? I have periodically been banging the drum about getting the HTCondor community off of an email list as the means of answering technical questions and onto something more modern. Rather than setting up something from scratch, piggybacking on an existing effort seems promising.

**Reply by vsoch** [reply]
It should be - I stopped participating in the weekly meetings something in 2019 but I spoke with Julie earlier this year and they are still charging forward on it. I think if you wanted to help and had vision for the resource they would greatly appreciate it! You could probably (akin to how they make spaces for groups) get an HTCondor space. I forget what they are called - "locales" maybe?

**Reply by alansill** [reply]
I have a YouTube space reserved in case we want to host original or donated content, duly vetted. We're also discussing something as above along the lines of a community index of separately maintained content. Opinions welcome from my point of view.

**Reply by mtwest2718** [reply]
I will ping Julie and see what I can do.

Getting old men to give up their beloved email lists is a fools errand but I can at least document/summarize stuff in a friendly space.

**Reply by mtwest2718** [reply]
I have a giant pile of topics in my head that would be wonderful, particularly for new infrastructure people, that would make good workshop ideas. Cover at least a few aspects of a single topic in one day. Preferable with hands-on assisted exercises. Yes, this does hint at the dreaded **standardization** and so be it.

- The chaotic symphony of batch scheduler daemons
- How to live without shared file system
- Making sense of networking nightmare
- The pains and pleasures of virtualization
- Who funds your infrastructure and how to be their friend
- Build your first (multi-node) cluster from start to finish
- Cluster visualization, the birds-eye view of your resources and their usage

**Reply by vsoch** [reply]
What about carpentries? Could you offer to start working on a lesson over there and get the community involved here?

**Comment by hearnsj** [comment]
Wow. And here we are - almost first hit on HPC Reddit is a cry for help on xcat courses.

https://www.reddit.com/r/HPC/comments/z23z4z/xcat_education/

For me, my learning path for xct was to get stuck in on an installation and to use the guides from OpenHPC plus the online manuals. However I have enough experience to know how to find things out in manuals like that and the confidence to go ahead knowing that I won't make irreversible changes. The cub HPC admin will not have confidence like that.

(If using the term 'cub' is offensive I will happily retract that. But I do like it)

**Reply by vsoch** [reply]
Yes very funny that the first post is someone saying "help meee!" I hope all of the help guides aren't buried in actual install manuals though - there are some posts on there that have good links to resources, e.g.,

![image](https://user-images.githubusercontent.com/814322/204422955-e8b1fbc8-5652-4636-9d10-53b2ce9dac15.png)

and historically I've seen a few that are "help me setup my first cluster" and the comments have useful advice (I think?)

**Reply by mtwest2718** [reply]
cub is kinda weird TBH
i call myself a newb, which itself has its own baggage
"new HPC admin" is perfectly adequate for most contexts I imagine.

**Comment by sramkrishna** [comment]
Don't forget to use something like asciinema to record tutorials if you're using a terminal on Linux. It basically like the old unix 'script' command, but it uses CSS to do the recording. You can then play it back and the best part is that it is like a video you can pause, rewind, pause, fast-forward but best of all you can cut-n-paste the commands and then put it on your screen and compare what shows up on the video and what you have. 

https://asciinema.org/

**Reply by vsoch** [reply]
Haha I'm so old school on that service my entire screen name is just "vs" https://asciinema.org/~vs :P

**Reply by vsoch** [reply]
This is my fave: https://asciinema.org/a/258360

**Comment by mtwest2718** [comment]
When it comes to tutorials, I feel like the SysAdmin/DevOps community would greatly benefit from explicit on-boarding lessons. And TBC, I don't mean talks but actual lessons. Like you would get at a workshop walk-through focusing on a narrow set of skills.

**Reply by vsoch** [reply]
I don't think I could be of much help there, I try to stay away from HPC as much as I can, ironically enough :laughing:

**Comment by sramkrishna** [comment]
I did too - and I see their point. Have a large general programming
community and then use flags or whatever.

**Comment by sramkrishna** [comment]
We in the GNOME community completely got rid of mailing lists this year and
we are now 100% on Discourse. You can set up a "mailing list" mode - but it
makes it easy to moderate.

**Comment by sramkrishna** [comment]
Oh yeah, i agree - although in the community I am in - it was late 30s and
40 year olds. But it's terrible for us because there is so much spam to
deal with.

**Reply by vsoch** [reply]
I think all of us here are in that same age bucket 😆

**Comment by sramkrishna** [comment]
Oh no.. not I. I think I'm a decade past lol.
